packages/block-library/src/search/style.scss

```diff
@@ -68,6 +68,12 @@
 			min-width: 0 !important;
 		}
 
+		.wp-block-search__input {
+			transition-property: all;
```

Comment:
This needs to be `all` and not `flex-basis` because using `flex-basis` some versions of Safari don't properly apply the transition.